pymc/distributions/timeseries.py

```diff
@@ -33,6 +36,137 @@
 ]
 
 
+class GaussianRandomWalkRV(RandomVariable):
+    """
+    GaussianRandomWalk Random Variable
+    """
+
+    name = "GaussianRandomWalk"
+    ndim_supp = 1
+    ndims_params = [0, 0, 0]
+    dtype = "floatX"
+    _print_name = ("GaussianRandomWalk", "\\operatorname{GaussianRandomWalk}")
+
+    def _shape_from_params(self, dist_params, reop_param_idx=0, param_shapes=None):
+        steps = dist_params[2]
+
+        # TODO: Ask ricardo why this is correct. Isn't shape different if size is passed?
+        return (steps,)
+
+    @classmethod
+    def rng_fn(
+        cls,
+        rng: np.random.RandomState,
+        mu: Union[np.ndarray, float],
+        sigma: Union[np.ndarray, float],
+        steps: int,
+        size: Tuple[int],
+    ) -> np.ndarray:
+        """Gaussian Random Walk generator.
+
+        The init value is drawn from the Normal distribution with the same sigma as the
+        innovations.
+
+        Notes
+        -----
+        Currently does not support custom init distribution
+
+        Parameters
+        ----------
+        rng: np.random.RandomState
+           Numpy random number generator
+        mu: np.ndarray
```

Comment:
this should be `array_like` instead. It won't match the type hints but it will match the actual accepted types, they are passed as is to the numpy function